src/Http/Http/src/Internal/ResponseCookies.cs

```diff
@@ -167,9 +152,77 @@ public void Delete(string key, CookieOptions options)
         });
     }
 
+    private bool TryGetLogger([NotNullWhen(true)] out ILogger? logger)
+    {
+        if (!_retrievedLogger)
+        {
+            _retrievedLogger = true;
+            var services = _features.Get<IServiceProvidersFeature>()?.RequestServices;
+            _logger = services?.GetService<ILogger<ResponseCookies>>();
+        }
+
+        logger = _logger;
+        return logger is not null;
+    }
+
+    private static MessagesToLog GetMessagesToLog(CookieOptions options)
+    {
+        var toLog = MessagesToLog.None;
+
+        if (!options.Secure && options.SameSite == SameSiteMode.None)
+        {
+            toLog |= MessagesToLog.SameSiteNotSecure;
+        }
+
+        if (!options.Secure && options.Partitioned)
+        {
+            toLog |= MessagesToLog.PartitionedNotSecure;
+        }
+
+        if (options.Partitioned && options.SameSite != SameSiteMode.None)
+        {
+            toLog |= MessagesToLog.PartitionedNotSameSiteNone;
+        }
```

Comment:
Should we also log if `options.Path != "/"`? It looks like just as with non-Secure cookies, it will get rejected.

https://developers.google.com/privacy-sandbox/blog/chips-origin-trial#cookies_requirements

Reply:
I implemented the restrictions from Barry's issue - I haven't actually read the CHIPS document yet.  There may well be other validation to perform.

Reply:
These restrictions feel like part of the spec that isn't baked yet.  But updating our validation and logging is much less problematic than renaming the properties.

Reply:
I don't see any anticipation of such a restriction in either the GH write-up or the draft spec modification.  Maybe log it as info/debug and say "some clients"?

Reply:
Logged with lower priority.